lib/internal/url.js

```diff
@@ -1443,6 +1444,8 @@ function encodePathChars(filepath) {
 }
 
 function pathToFileURL(filepath) {
+  validateString(filepath, 'path');
```

Comment:
This is an extremely hot path, hence my comment:

We already validate this for `!(isWindows && StringPrototypeStartsWith(filepath, '\\\\'))` case (path.resolve handles the validation). Can you make sure this only runs when the OS is windows? No need to run it twice for  non-windows environments.

Reply:
This check also fixes `The "paths[0]" argument must be of type string.` error message that kinda implies that this method may take more than one argument.

I guess we can keep the `internal/url` version as is (so it won't affect performance in core), but instead of re-exporting it directly in `url` make a thin wrapper that validates input, wdyt?

Reply:
That sounds good to me.